src/background/javascript-instrument.ts

```diff
@@ -1,54 +1,62 @@
-    update.time_stamp = data.timeStamp;
-
-    // document_url is the current frame's document href
-    // top_level_url is the top-level frame's document href
-    update.document_url = loggingDB.escapeString(sender.url);
-    update.top_level_url = loggingDB.escapeString(sender.tab.url);
-
-    // Create a json object for function arguments
-    // We create an object that maps array positon to argument
-    // e.g. someFunc('a',123,'b') --> {0: a, 1: 123, 2: 'b'}
-    // to make it easier to query the data, using something like the
-    // sqlite3 json1 extension.
-    const args = {};
-    if (data.operation === "call" && data.args.length > 0) {
-      for (let i = 0; i < data.args.length; i++) {
-        args[i] = data.args[i];
-      }
-      update.arguments = loggingDB.escapeString(JSON.stringify(args));
-    }
+import { escapeString } from "../lib/string-utils";
+import { JavascriptOperation } from "../types/schema";
+
+export class JavascriptInstrument {
+  private readonly dataReceiver;
 
-    loggingDB.saveRecord("javascript", update);
+  constructor(dataReceiver) {
+    this.dataReceiver = dataReceiver;
   }
 
-  // Listen for messages from content script injected to instrument JavaScript API
-  browser.runtime.onMessage.addListener(function(msg, sender, sendReply) {
-    // console.debug("javascript-instrumentation background listener - msg, sender, sendReply", msg, sender, sendReply);
-    if (msg.namespace && msg.namespace === "javascript-instrumentation") {
-      switch (msg.type) {
-        case "logCall":
-        case "logValue":
-          processCallsAndValues(msg.data, sender);
-          break;
+  public run(crawlID) {
+    console.log("JavascriptInstrument", crawlID, this.dataReceiver);
+
+    const processCallsAndValues = (data, sender) => {
+      const update = {} as JavascriptOperation;
+      update.crawl_id = crawlID;
+      update.script_url = escapeString(data.scriptUrl);
+      update.script_line = escapeString(data.scriptLine);
+      update.script_col = escapeString(data.scriptCol);
+      update.func_name = escapeString(data.funcName);
+      update.script_loc_eval = escapeString(data.scriptLocEval);
+      update.call_stack = escapeString(data.callStack);
+      update.symbol = escapeString(data.symbol);
+      update.operation = escapeString(data.operation);
+      update.value = escapeString(data.value);
+      update.time_stamp = data.timeStamp;
+
+      // document_url is the current frame's document href
+      // top_level_url is the top-level frame's document href
+      update.document_url = escapeString(sender.url);
+      update.top_level_url = escapeString(sender.tab.url);
+
+      // Create a json object for function arguments
+      // We create an object that maps array positon to argument
+      // e.g. someFunc('a',123,'b') --> {0: a, 1: 123, 2: 'b'}
+      // to make it easier to query the data, using something like the
+      // sqlite3 json1 extension.
```

Comment:
We can remove the `sqlite3 json1` recommendation since this isn't tied to sqlite anymore

Reply:
Done and noted as a breaking change